crates/ide_assists/src/handlers/extract_function.rs

```diff
@@ -1441,21 +1440,31 @@ fn make_body(
         FunctionBody::Span { parent, text_range } => {
             let mut elements: Vec<_> = parent
                 .syntax()
-                .children()
+                .children_with_tokens()
                 .filter(|it| text_range.contains_range(it.text_range()))
-                .map(|it| rewrite_body_segment(ctx, &fun.params, &handler, &it))
+                .map(|it| match &it {
+                    syntax::NodeOrToken::Node(n) => syntax::NodeOrToken::Node(
+                        rewrite_body_segment(ctx, &fun.params, &handler, &n),
+                    ),
+                    _ => it,
+                })
                 .collect();
```

Comment:
I fear this might allow one to partially extract parts of a block, that is
```rs

fn foo() {
    $0
    foo();
}$0
```
might work now, tearing the body apart by moving one of the braces

Reply:
You're right, this was breaking. I've added a unit test to guard against this in the future